presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java

```diff
@@ -332,6 +332,11 @@ public HiveWriter createWriter(Page partitionColumns, int position, OptionalInt
                     updateMode = UpdateMode.NEW;
                     writeInfo = locationService.getPartitionWriteInfo(locationHandle, partition, partitionName.get());
                 }
+                else if (table.isTemporary()) {
+                    // Note: temporary table is always empty at this step
+                    updateMode = UpdateMode.APPEND;
+                    writeInfo = locationService.getTableWriteInfo(locationHandle);
```

Comment:
nit: check the write mode is `DIRECT_TO_TARGET_EXISTING_DIRECTORY`

Reply:
We may change it in a very next iteration. I don't think it makes much sense to check it now.